src/ngraph/runtime/cpu/pass/cpu_mat_fusion.cpp

```diff
@@ -327,21 +327,35 @@ bool runtime::cpu::pass::CPURnnMatFusion::run_on_function(std::shared_ptr<Functi
             auto add_node = std::make_shared<op::Add>(dot_node, bias_broadcast_node);
             const auto& add_shape = add_node->get_shape();
 
+            // we will sort the captured Add(Dot(X, W) + B) as per the the slice ordering of X
+            // this will simplify the replace_node logic
+            auto lambda = [&](const std::shared_ptr<Node> node1,
+                              const std::shared_ptr<Node> node2) {
+                const auto node1_slice =
+                    std::static_pointer_cast<op::Slice>(op_seg_map[node1].at(Type::DATA));
+
+                const auto node2_slice =
+                    std::static_pointer_cast<op::Slice>(op_seg_map[node2].at(Type::DATA));
+
+                return (node1_slice->get_lower_bounds() < node2_slice->get_lower_bounds() &&
+                        node1_slice->get_upper_bounds() < node2_slice->get_upper_bounds());
+            };
+            std::sort(op_nodes.begin(), op_nodes.end(), lambda);
```

Comment:
@Krovatkin not needed inside the lambda, since it is a quick sort..their will be @least one pair where comparison results in `false`. I think we can do the comparison after sorted , but its not needed.

Reply:
@pruthviIntel yup, you are right. that check needs to be done on a sorted sequence.